### PR TITLE
[new-app] Morteza/routing

### PR DESCRIPTION
**Pull request by morteza-binary**

### CODE REVIEW

File: 404.html [hunk 1]
```diff
@@ -84,27 +84,32 @@
      */
     function handleReactUrl() {
 
+        var l = window.location;
+
         // if the url is not related to new app, returns false.
-        var lang_regex_string = '(' + all_languages.join('|') + ')\/app\.html';
+        var lang_regex_string = '(.*)\/(' + all_languages.join('|') + ')\/app\/(\\w+\\.html)?';
+        var rxResult = new RegExp(lang_regex_string).exec(l.pathname)
```
Comment (binary-hamedanchi) [inline 404.html]: Probably `rx_result`? 😄

File: 404.html [hunk 2]
```diff
@@ -84,27 +84,32 @@
...
             return false;
         }
 
-        // Since there are two sub entries after CNAME URL for the app 'www.binary.com/en/app.html'
+        var sub_directories = rxResult[1];
+
+        // Since there are two sub entries after CNAME URL for the app 'www.binary.com/en/app'
         // the value of segment_count should be 2.
-        var segment_count = 2;
+        var segment_count =  2 + sub_directories.split('/').length;
```
Comment (negarn) [inline 404.html]: extra whitespace before `2`

File: src/javascript/app_2/App/app.js [hunk 3]
```diff
@@ -3,9 +3,10 @@ import PropTypes                     from 'prop-types';
 import React                         from 'react';
 import { render }                    from 'react-dom';
 import { BrowserRouter as Router }   from 'react-router-dom';
-import Footer                        from '../App/Containers/Layout/footer.jsx';
 import Header                        from './Components/Layout/header.jsx';
-import { BinaryRoutes }              from './routes';
+import BinaryRoutes                  from './Components/Routes';
+import Footer                        from '../App/Containers/Layout/footer.jsx';
```
Comment (kellybinary) [inline src/javascript/app_2/App/app.js]: footer path can be updated to `./Containers/Layout/..` 😄

File: src/javascript/app_2/Constants/routes.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const routes = {
+    account_password: '/account_password',
+    apps            : '/apps',
+    cashier_password: '/cashier_password',
+    exclusion       : '/exclusion',
+    financial       : '/financial',
+    history         : '/history',
+    index           : '/index',
+    limits          : '/limits',
+    personal        : '/personal',
```
Comment (binary-hamedanchi) [inline src/javascript/app_2/Constants/routes.js]: Is it possible to be more explicit here?
i.e. `/personal` finally would become `/settings/personal`, but the way stated here is unclear and confusing.

Comment (morteza-binary) [reply src/javascript/app_2/Constants/routes.js]: @binary-hamedanchi 
it's possible.

**Pull request merged**